### DISCUSSION TITLE
How to manage Repository Access via REST or CLI

### DISCUSSION THREAD

**Discussion by chris-pateman** [community/community#36528] [Apps, API and Webhooks] [answered]
I am trying to list all users and/or teams that have access to repositories.

I can run the  REST command to get all Teams >https://docs.github.com/en/rest/repos/repos#list-repository-teams

However, I also want to get what users have direct access as well, same as the https://github.com/ORG/REPO/settings/access page

Does anyone know of what CLI or REST command can do this?

**Comment by LangLangBart** [comment, marked as answer]
[GitHub Docs - collaborators](https://docs.github.com/en/rest/collaborators/collaborators#list-repository-collaborators) ?

**Reply by chris-pateman** [reply]
These, as far as I understand it, are people who have contributed to the repo, but no the people who have direct access to the repo.
When i call this I get all users.

**Reply by LangLangBart** [reply]
> These, as far as I understand it, are people who have contributed to the repo

collaborators != contributors

> List repository collaborators
[GitHub Docs - collaborators](https://docs.github.com/en/rest/collaborators/collaborators#list-repository-collaborators) 

> List repository contributors
[GitHub Docs - contributors](https://docs.github.com/en/rest/repos/repos#list-repository-contributors)

### Query parameters ?

> [...]
> permission
> Filter collaborators by the permissions they have on the repository. If not specified, all collaborators will be returned.

```zsh
# get all collaborators with admin permission
gh api -X GET repos/hmcts/spring-boot-template/collaborators -f permission=admin
```

**Reply by chris-pateman** [reply]
Ah you are correct thank you.
The thing throwing me off was the 30 users that came back was not on the page.
These are users with indirect access to the repo, but after adding `affiliation=direct` I get the list I need.

Thank you,

**Comment by ossanna16** [comment]
Hey there @chris-pateman and thank you for asking a good question! I'm glad you found our community 🙂

**Reply by LangLangBart** [reply]
> Hey there @chris-pateman and thank you for asking a good question! I'm glad you found our community 🙂

a nice gesture, but on the whole unnecessary noise without any contribution to the actual question.
